prow/github/report/report.go

```diff
@@ -156,7 +159,16 @@ func Report(ghc GithubClient, reportTemplate *template.Template, pj prowapi.Prow
 	if len(refs.Pulls) == 0 {
 		return nil
 	}
-
+	icomm, err := ghc.ListPRCommits(refs.Org, refs.Repo, refs.Pulls[0].Number)
+	if err != nil {
+		return fmt.Errorf("error listing commits: %v", err)
+	}
+	// Check if the ProwJob SHA exists in the list of PR commit SHAs.
+	// If it doesn't, skip reporting as this is an old ProwJob.
+	sha := pj.Spec.Refs.Pulls[0].SHA
+	if !findSHAInPRCommits(sha, icomm) {
```

Comment:
this logic needs a minor change.

it has to also handle the case where a commit exists in the list of commits in a PR, but this commit is not the latest.
in such a case skip too.

Reply:
https://github.com/kubernetes/test-infra/blob/6f40a71acfc801ca4c1f4a314f2293398487f576/prow/github/client.go#L2426

Reply:
just spoke to @BenTheElder about this during the kind meeting. he reminded me that this is by author date. so the handling won't be ideal and we have to rely on what the user pushes chronologically.

Reply:
at least, I'm pretty sure most of the APIs are by author date, @cjwagner is the expert 😅

Reply:
so i did a quick `GET` check and the github API for `pulls/N/commits` returns the same order as the order which can be viewed on the UI. in such an order the last commit in the list seems to  always be the one being tested.